L1TGlobalPrescaler: implement arbitrary prescales on top of the existing L1 uGT results

fwyzard opened the pull request:
The new L1T results will contain only the L1 decision for bx 0, as trying to simulate the effect of prescales across multiple bunch crossings is unlikely to give consistent results.

fwyzard committed:
L1TGlobalPrescaler: implement arbitrary prescales on top of the existing L1 uGT results

fwyzard commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
A new Pull Request was created by  @fwyzard (Andrea Bocci) for CMSSW_8_0_X.

It involves the following packages:

L1Trigger/L1TGlobal


@cmsbuild, @rekovic, @mulhearn, @davidlange6 can you please review it and eventually sign? Thanks.
@Martin-Grunewald this is something you requested to watch as well.
@slava77, @degano, @smuzaffar you are the release manager for this.

cms-bot commands are list here cms-sw/cmssw#13028

cmsbuild commented:
Comparison not run due to Build errors (RelVals and Igprof tests were also skipped)

cmsbuild commented:
-1

Tested at: d7680b20def6b4eddcdb552ca75a1b4cfead57d7

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-14502/12987/summary.html

I found follow errors while testing this PR

**Failed tests:  Build**

* **Build**:

I found an error when building: 
 
 <pre>gmake[1]: Target 'PostBuild' not remade because of errors.
gmake[1]: Leaving directory '/build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_0_X_2016-05-12-2300'
config/SCRAM/GMake/Makefile.rules:1935: recipe for target 'src' failed
gmake: *** [src] Error 2
gmake: Target 'all' not remade because of errors.
gmake: *** [There are compilation/build errors. Please see the detail log above.] Error 2
</pre>

fwyzard commented:
@davidlange6 the error is not in the module, but is due to `edmWriteConfig` generating incorrect python for long (> 255 elements) vdouble objects.

How can I fix that ?

fwyzard committed:
edmWriteConfig: fix python for large vectors

Python does not support function calls with more than 255 arguments:

    f(arg1, arg2, ..., arg255, ...)

The workaround is to wrap them in a tuple and expand it in place:

    f(*(arg1, arg2, ..., arg255, ...))

Add a check to `ParameterDescription.cc` so that `edmWriteConfig` can use
the latter syntax when generating `cfi.py` files with large vectors.

fwyzard commented:
@Dr15Jones could you review and sign the ParameterDescription.cc fix ?

fwyzard commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
Pull request #14502 was updated. @smuzaffar, @Dr15Jones, @cmsbuild, @rekovic, @mulhearn, @davidlange6 can you please check and sign again.

cmsbuild commented:
+1
Tested at: 682e613c945a285ca193e825bbd17ed2541c460f
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-14502/13013/summary.html

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-14502/13013/summary.html

Dr15Jones commented:
+1
Looks OK to me. Would be even better if one of the test cases was extended to check the new code :).

fwyzard commented:
>  Looks OK to me. Would be even better if one of the test cases was extended to check the new code :).
I'll leave that to the framework people... I can confirm that the new code produces the correct python for `l1tGlobalPrescaler_cfi.py` :-)
